scripts/generate_black_spots_input.py

```diff
@@ -0,0 +1,374 @@
+            'records': len(records) if records else 0
+        }
+
+        # Add time offset aggregation data
+        for year_range in year_ranges:
+            max_occurred, min_occurred, records_label, severe_label = year_range
+            if records:
+                records_in_range = [
+                    record for record in records
+                    if min_occurred < record['occurred'] <= max_occurred
+                ]
+                severe_records_in_range = [
+                    record for record in records_in_range
+                    if record['severe']
+                ]
+                data[records_label] = len(records_in_range)
+                data[severe_label] = len(severe_records_in_range)
+            else:
+                data[records_label] = 0
+                data[severe_label] = 0
+
+        segments_with_data.append((segment, data))
+
+    return (schema, segments_with_data)
+
+
+def write_segments_shp(segments_shp_path, road_projection, segments_with_data, schema):
+    """Writes all segments to shapefile (both intersections and individual segments)"""
+    with fiona.open(segments_shp_path, 'w', driver='ESRI Shapefile',
+                    schema=schema, crs=road_projection) as output:
+        for segment_with_data in segments_with_data:
+            segment, data = segment_with_data
+            output.write({
+                'geometry': mapping(segment),
+                'properties': data
+            })
+
+
+def write_segments_csv(segments_csv_path, segments_with_data, schema):
+    """Writes all segments containing record data to csv"""
+    field_names = sorted(schema['properties'].keys())
+    with open(segments_csv_path, 'w') as csv_file:
+        csv_writer = csv.DictWriter(csv_file, fieldnames=field_names)
+        csv_writer.writeheader()
+        for segment_with_data in segments_with_data:
+            _, data = segment_with_data
+            if data['records'] > 0:
+                csv_writer.writerow(data)
+
+
+def main():
+    parser = argparse.ArgumentParser(description='Generate black spots input')
+    # Required arguments
+    parser.add_argument('roads_shp', help='Path to shapefile containing road data')
+    parser.add_argument('records_csv', help='Path to CSV containing record data')
+
+    # Optional arguments
+    parser.add_argument('--output-dir', help='Directory where files are output', default='.')
+    parser.add_argument('--combined-segments-shp-name', help='Combined segments output .shp name',
+                        default='combined_segments.shp')
+    parser.add_argument('--combined-segments-csv-name', help='Combined segments output .csv name',
+                        default='combined_segments.csv')
+    parser.add_argument('--intersection-buffer-meters', help='Meters to buffer each intersection',
+                        default=5)
+    parser.add_argument('--max_line_meters', help='Maximum meters allowed for line segment',
```

Comment:
That's true, there's no reason it needs to be in meters. I can switch it to `units` or something similar in case someone wants to use something like feet.